lib/hexpm/web/router.ex

```diff
@@ -119,6 +95,39 @@ defmodule Hexpm.Web.Router do
     get "/blog/:slug", BlogController, :show
   end
 
+  scope "/dashboard", Hexpm.Web.Dashboard do
+    pipe_through :browser
+
+    get "/profile", ProfileController, :index
+    post "/profile", ProfileController, :update
+
+    get "/password", PasswordController, :index
+    post "/password", PasswordController, :update
+
+    get "/email", EmailController, :index
+    post "/email", EmailController, :create
+    delete "/email", EmailController, :delete
+    post "/email/primary", EmailController, :primary
+    post "/email/public", EmailController, :public
+    post "/email/resend", EmailController, :resend_verify
+    post "/email/gravatar", EmailController, :gravatar
+
+    get "/repos/:dashboard_repo", RepositoryController, :show
+    post "/repos/:dashboard_repo", RepositoryController, :update
+    post "/repos/:dashboard_repo/billing-token", RepositoryController, :billing_token
+    post "/repos/:dashboard_repo/cancel-billing", RepositoryController, :cancel_billing
+    post "/repos/:dashboard_repo/update-billing", RepositoryController, :update_billing
+    post "/repos/:dashboard_repo/create-billing", RepositoryController, :create_billing
+    get "/repos/:dashboard_repo/invoices/:id", RepositoryController, :show_invoice
+    post "/repos/:dashboard_repo/invoices/:id/pay", RepositoryController, :pay_invoice
+    get "/repos", RepositoryController, :new
+    post "/repos", RepositoryController, :create
+
+    get "/keys", KeyController, :index
+    delete "/keys", KeyController, :delete
```

Comment:
Hi @ericmj Great to see you separating the user dashboard related entities into separate controllers 👍. Out of curiosity, can you please throw some light on why is the revoking of a key is an HTTP `delete` and not a `patch` or `put` ?

I was looking at the logs and what I found it doing was the below for a user key named `test-2`:

```
[debug] QUERY OK db=1.6ms
UPDATE "keys" SET "revoked_at" = $1, "updated_at" = $2 WHERE "id" = $3 [{{2018, 5, 12}, {7, 49, 10, 217092}}, {{2018, 5, 12}, {7, 49, 10, 217559}}, 36]
[debug] QUERY OK db=1.7ms
INSERT INTO "audit_logs" ("action","actor_id","params","user_agent","inserted_at") VALUES ($1,$2,$3,$4,$5) RETURNING "id" ["key.remove", 5, %{id: 36, name: "test-2", permissions: [%{domain: "api", resource: nil}]}, "WEB", {{2018, 5, 12}, {7, 49, 10, 219359}}]
[debug] QUERY OK db=0.5ms
commit []
[debug] QUERY OK source="sessions" db=0.9ms
UPDATE "sessions" AS s0 SET "data" = $1, "updated_at" = $2 WHERE (s0."id" = $3) [%{"_csrf_token" => "Dt/P5s2s5L6d24fbtGSHtw==", "phoenix_flash" => %{"info" => "The key test-2 was revoked successfully"}, "user_id" => 5}, {{2018, 5, 12}, {7, 49, 10, 222070}}, 2]
[info] Sent 302 in 15ms
```

From the logs, it looks like it's doing an update and not a delete hence my question. Please feel free to correct me if I'm missing something. I'm more curious about this because it's on the basis of this I had used `patch` as part of my PR changes [here](https://github.com/hexpm/hexpm/pull/670/files#diff-4dafadbd5feb6976b0fe65de241a2498R94). It would be great to understand this better. Thank you!

Reply:
It's an update internally but that's an implementation detail. Semantically, for the REST interface, it behaves like a delete.

Reply:
Thanks for sharing why you've considered using `delete` here @ericmj.